src/App.tsx

```diff
@@ -15,36 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [sortMode, setSortMode] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const sorteredGoods: string[] = [...goodsFromServer].sort((a, b) => {
+    if (sortMode === SortType.Alphabetically) {
+      return a.localeCompare(b);
+    } else if (sortMode === SortType.Length) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  const finalGoods: string[] = isReverse
+    ? [...sorteredGoods].reverse()
+    : sorteredGoods;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${!(sortMode === SortType.Alphabetically) && 'is-light'}`}
+          onClick={() => {
+            setSortMode(SortType.Alphabetically);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${!(sortMode === SortType.Length) && 'is-light'}`}
+          onClick={() => {
+            setSortMode(SortType.Length);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${!isReverse && 'is-light'}`}
+          onClick={() => {
+            setIsReverse(!isReverse);
+          }}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortMode !== SortType.Default || isReverse) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReverse(false);
+              setSortMode(SortType.Default);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {finalGoods.map((good: string) => {
```

Comment:
You don't have to specify type every time, typescript understands what type it has